### PR TITLE
Fixes #21222: Improve yaml format

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/21222

### CODE REVIEW

File: rudderc/src/ir/resource.rs [hunk 1]
```diff
@@ -18,23 +19,43 @@ pub struct Policy {
...
     // comes from stdlib
     pub report_parameter: String,
-    // named params?
-    pub params: Vec<String>,
+    pub state_type: String,
 }
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct ReportingPolicy {
+    pub enabled: Option<bool>,
```
Comment (amousset) [inline rudderc/src/ir/resource.rs:46]: I think these parameters shouldn't be `Option` but have a default value instead.

Comment (amousset) [reply rudderc/src/ir/resource.rs:46]: With https://serde.rs/field-attrs.html#default--path or https://serde.rs/field-attrs.html#default

File: rudderc/src/ir/resource.rs [hunk 2]
```diff
@@ -18,23 +19,43 @@ pub struct Policy {
...
+pub enum ReportingCompute {
+    #[serde(rename = "worst-weighted-sum")]
+    WorstCaseWeightedSum,
+    #[serde(rename = "worst-one")]
+    WorstCaseWeightedOne,
+    #[serde(rename = "focus")]
+    Focus(String),
+    #[serde(rename = "default")]
+    Weighted
+}
```
Comment (amousset) [inline rudderc/src/ir/resource.rs:61]:
```suggestion
}

impl Default for ReportingCompute {
  fn default() -> Self {
    ReportingCompute::Weighted
  }
}
```

File: rudderc/src/ir/resource.rs (hunk 1 above)
Comment (Fdall) [reply rudderc/src/ir/resource.rs:46]: :+1:

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
this is missing a `cargo fmt`

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Comment by amousset** [comment]
Tests fail due to the yaml in comments. We can remove it for now.

**Pull request closed without merging**